agent/node.go

```diff
@@ -607,8 +607,9 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 			return err
 		}
 		done := make(chan struct{})
+		var runErr error
 		go func() {
-			m.Run(context.Background()) // todo: store error
+			runErr = m.Run(context.Background())
```

Comment:
The fact that this is `context.Background()` has me worried. Why isn't this using the function's context?